### ISSUE TITLE
eval: tool-calling SLM tournament and Agent0-style self-evolution track

### ISSUE THREAD

**Issue by kvnloo** [kvnloo/evolution-lab#23]
## Goal

Extend the existing observer-first evaluation work in #20 into a reproducible tournament for local tool-calling/orchestration policies, then test whether Agent0-style self-evolution can improve the winning local policy on our own traces.

Runtime inventory/control belongs in https://github.com/kvnloo/z0intelligence/issues/20. Evolution Lab owns grouped experiments, frozen evaluation, training/search, and promotion evidence.

## Candidates

Evaluate at least:

- Nemotron-Orchestrator-8B
- Qwen3.5-9B
- Qwen3.5-4B
- Hammer2.1-3B
- Hammer2.1-7B where useful
- FunctionGemma-270M after task-specific fine-tuning
- JEV/OpenJev bounded scorer
- deterministic/rules/logistic controls

Agent0 is a separate **training methodology arm**, not just another checkpoint.

## Benchmark the actual control problem

Do not reduce this to JSON validity or one function-calling leaderboard.

For frozen work-item groups, measure:

```text
verified task success
correct tool/model selection
invalid call rate
unnecessary call rate
dependency/order violations
tool irrelevance rejection
parallelization quality
retry count
recovery success
escalation precision/recall
stop/continue quality
wall-clock latency
TTFT
decision latency
GPU-ms
VRAM peak
tokens
dollar cost
calibration / abstention
```

Keep syntax/function-call accuracy and trajectory/control success as separate axes.

## A/B matrix

Start with:

A. Qwen3.5-9B alone  
B. Nemotron-Orchestrator-8B alone  
C. deterministic legal-action compiler + orchestrator  
D. compiler + JEV + orchestrator  
E. compiler + JEV + orchestrator + Qwen fallback  
F. same stack + distilled fly/mushroom/tiny specialist fast paths

Run 4B/3B/270M substitutions where the role permits.

## Agent0-style track

Use verified traces to explore curriculum/executor co-evolution:

- generate harder tool-aware tasks from real failure clusters
- train executor policies against frozen verifiers
- prevent self-generated reward from becoming ground truth
- compare against DAgger/current autoresearch baselines
- export only candidates that clear frozen held-out gates

No online self-modification of production authority.

## Leakage / promotion

Preserve the grouping discipline from #20: retries, branches, continuations and repeated observations from one work item stay in one fold. Promotion is per role/question family, not “one best model overall.”

## Acceptance

- [ ] frozen local-tool-calling suite exists
- [ ] trajectory metrics are distinct from schema/syntax metrics
- [ ] 3080 Ti latency/VRAM measurements are captured
- [ ] deterministic/JEV/SLM compositions are compared, not only individual checkpoints
- [ ] cost-quality-latency Pareto frontier is exported
- [ ] Agent0-style training arm uses independent verifiers and held-out evaluation
- [ ] promotion evidence can be consumed by z0intelligence
- [ ] no security/capability authority is learned

Related: #20, https://github.com/kvnloo/z0intelligence/issues/20

**Comment by kvnloo** [comment]
Free-tier spillover can become useful experiment capacity once the protected routing path is in place.

Kerdoios tracker: https://github.com/kvnloo/kerdoios/issues/51

When real DSH/Hermes workload is insufficient to consume compatible Groq/Cerebras free quota before reset, Evolution Lab may offer **explicitly queued** eval/distillation/search work as low-priority fill. This must not generate meaningless filler solely to exhaust quota.

Keep the existing split discipline: protected confirm/OOD cases remain certification-only; background free-quota work should use training/validation or explicitly disposable research tasks.
